seismicpro/src/seismic_batch.py

```diff
@@ -882,6 +883,132 @@ def equalize(self, index, src, dst, params, survey_id_col=None, upscale=False):
+                                           samples_step=samples_step, window=window)
+        semblance_len = (upper_bounds - lower_bounds).max()
+        residual_semblance = np.zeros((len(times), semblance_len))
+
+        for i, (low_bound, up_bound) in enumerate(zip(lower_bounds, upper_bounds)):
+            ix = (semblance_len - (up_bound - low_bound)) // 2
+            residual_semblance[i, ix : ix + up_bound - low_bound] = semblance[i, low_bound : up_bound]
+
+        getattr(self, dst)[pos] = residual_semblance
+
+    @action
+    @inbatch_parallel(init='_init_component', target='threads')
+    @apply_to_each_component
+    def add_muting(self, index, src, dst, interp_type='interpolation', muting=None, picking=None, indent=None):
+        """ Zeroing seismogram below ``muting`` or ``picking`` times.
+
+        Parameters
+        ----------
+        some parameters
+
+        muting : array-like, optional
+            Array with structure:
+            ``[[time_1, offset_1],
+                ...
+               [time_N, offset_N]]``
+        Picking : str, optional
+            Name of the comonent with picking values.
+
+        """
+        pos = self.index.get_pos(index)
+        field = getattr(self, src)[pos]
+        offsets = np.sort(self.index.get_df(index=index)['offset'])
+        samples_step = np.diff(self.meta[src]['samples'][:2])[0]
+        data_x, data_y = None, None
+        indent = indent if indent is not None else 0
+
+        if picking is not None:
+            picking = getattr(self, picking)[pos]
+            data_y = np.array(picking, dtype=np.int32) # ms
+            data_x = offsets[:len(data_y)].copy() # meters
+        elif muting is not None:
+            indent = np.zeros(len(field))
+            muting = np.asarray(muting) if isinstance(muting, (tuple, list)) else muting
+            data_y, data_x = muting[:, 0], muting[:, 1]
+        else:
+            raise ValueError('Either `picking` or `muting` should be determined.')
+
+        if interp_type == 'interpolation':
+            poly = np.polyfit(data_x, data_y, deg=2)
+            mute_samples = np.polyval(poly, offsets) / samples_step
+            if np.sum(mute_samples < 0) > 0:
+                mute_samples[mute_samples < 0] = 0
+        elif interp_type == 'regression':
+            lin_reg = LinearRegression(fit_intercept=False)
+            lin_reg.fit(data_y.reshape(-1, 1), data_x)
+            indent /= 1000 # to ms
+            velocity = lin_reg.coef_ - indent # reduce velocity by given indent. lin_reg.coef_ has m / ms
+            mute_samples = offsets / (velocity * samples_step) # meters / samples
+            mute_samples = mute_samples.astype(int)
+        else:
+            raise ValueError('Interpolation type must be "interpolation" or "regression" not "{}"'.format(interp_type))
+        mute_time_mx = np.array([mute_samples]*field.shape[1]).T
```

Comment:
```suggestion
        mute_mask = (np.arange(field.shape[1]).reshape(1, -1) - mute_samples.reshape(-1, 1)) > 0
```
This way you get array of bools with shape `(mute_samples, field.shape[1])` and can proceed to line 1008
`muted_field = field * mute_mask`